tools/topology/topology2/platform/intel/dmic-generic.conf

```diff
@@ -62,180 +62,183 @@ Object.Dai.DMIC [
-			Object.Base.audio_format.1 {
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				out_bit_depth		32
-				out_valid_bit_depth	32
-			}
-			Object.Base.audio_format.2 {
-				in_channels		4
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				out_channels		4
-				out_bit_depth		32
-				out_valid_bit_depth	32
-				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
-				in_ch_map	$CHANNEL_MAP_3_POINT_1
-				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
-				out_ch_map	$CHANNEL_MAP_3_POINT_1
-			}
-		}
+				Object.Widget.copier.1 {
+					dai_index	0
+					dai_type	"DMIC"
+					copier_type	"DMIC"
+					type		dai_out
+					stream_name	$DMIC0_NAME
+					node_type $DMIC_LINK_INPUT_CLASS
+					num_input_audio_formats 2
+					num_output_audio_formats 2
+					Object.Base.audio_format.1 {
+						in_bit_depth		32
+						in_valid_bit_depth	32
+						out_bit_depth		32
+						out_valid_bit_depth	32
+					}
+					Object.Base.audio_format.2 {
+						in_channels		4
+						in_bit_depth		32
+						in_valid_bit_depth	32
+						out_channels		4
+						out_bit_depth		32
+						out_valid_bit_depth	32
+						in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+						in_ch_map	$CHANNEL_MAP_3_POINT_1
+						out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+						out_ch_map	$CHANNEL_MAP_3_POINT_1
+					}
+				}
 
-		Object.Widget.module-copier."2" {
-			stream_name $DMIC0_NAME
-			num_input_audio_formats 2
-			num_output_audio_formats 2
-			Object.Base.audio_format.1 {
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				out_bit_depth		32
-				out_valid_bit_depth	32
-			}
-			Object.Base.audio_format.2 {
-				in_channels		4
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				out_channels		4
-				out_bit_depth		32
-				out_valid_bit_depth	32
-				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
-				in_ch_map	$CHANNEL_MAP_3_POINT_1
-				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
-				out_ch_map	$CHANNEL_MAP_3_POINT_1
-			}
-		}
+				Object.Widget.module-copier."2" {
+					stream_name $DMIC0_NAME
+					num_input_audio_formats 2
+					num_output_audio_formats 2
+					Object.Base.audio_format.1 {
+						in_bit_depth		32
+						in_valid_bit_depth	32
+						out_bit_depth		32
+						out_valid_bit_depth	32
+					}
+					Object.Base.audio_format.2 {
+						in_channels		4
+						in_bit_depth		32
+						in_valid_bit_depth	32
+						out_channels		4
+						out_bit_depth		32
+						out_valid_bit_depth	32
+						in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+						in_ch_map	$CHANNEL_MAP_3_POINT_1
+						out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+						out_ch_map	$CHANNEL_MAP_3_POINT_1
+					}
+				}
 
-		Object.Widget.eqiir.1 {
-			num_input_audio_formats 2
-			num_output_audio_formats 2
-			Object.Base.audio_format.1 {
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				out_bit_depth		32
-				out_valid_bit_depth	32
+				Object.Widget.eqiir.1 {
+					num_input_audio_formats 2
+					num_output_audio_formats 2
+					Object.Base.audio_format.1 {
+						in_bit_depth		32
+						in_valid_bit_depth	32
+						out_bit_depth		32
+						out_valid_bit_depth	32
```

Comment:
does this need to be split?

Reply:
@plbossart this is existing code that's being made optional. We should absolutely split it but not in this patch. I have a series coming up for cleaning this up.